### PR TITLE
Split editor setup screen up into sections

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #10388 to avoid very bad merge conflicts.

Unlike the [design](https://user-images.githubusercontent.com/1329837/42362632-3042c92e-812f-11e8-8ee8-85ad31f83745.png) implies, I believe we want all the beatmap settings to be in a single scroll view like the user profile content.

I started to look at sharing implementation with `UserProfileOverlay` to get the tab control displaying but this requires some abstraction work (it's currently very local to that class). Definitely worth putting in the effort to abstract this as it is used in multiple other places on osu-web already.

### CODE REVIEW

File: osu.Game/Screens/Edit/Setup/SetupSection.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics;
+using osuTK;
+
+namespace osu.Game.Screens.Edit.Setup
+{
+    internal class SetupSection : Container
+    {
+        protected FillFlowContainer Flow;
```
Comment (smoogipoo) [inline osu.Game/Screens/Edit/Setup/SetupSection.cs:17]:
```suggestion
        protected FillFlowContainer Flow { get; private set; }
```

File: osu.Game/Screens/Edit/Setup/SetupSection.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        [Resolved]
+        protected OsuColour Colours { get; private set; }
+
+        [Resolved]
+        protected IBindable<WorkingBeatmap> Beatmap { get; private set; }
+
+        public override void Add(Drawable drawable) => throw new InvalidOperationException("Use Flow.Add instead");
```
Comment (smoogipoo) [inline osu.Game/Screens/Edit/Setup/SetupSection.cs:25]: Is there a reason why this is not set as override of `Content`?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Edit/Setup/SetupSection.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Screens/Edit/Setup/SetupSection.cs:25]: FillFlowContainer ain't a Container :(

Comment (smoogipoo) [reply osu.Game/Screens/Edit/Setup/SetupSection.cs:25]: Oh, you might need to use `FillFlowContainer<Drawable>`. I'm 99% sure this works.

Comment (smoogipoo) [reply osu.Game/Screens/Edit/Setup/SetupSection.cs:25]: Actually no, am I misunderstanding something? It should just work by the looks of it:
image

Comment (peppy) [reply osu.Game/Screens/Edit/Setup/SetupSection.cs:25]: Weird, it does work. I may have changed the type since I determined it didn't.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**